projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cds/profile-tag.e2e-spec.ts

```diff
@@ -17,22 +20,28 @@ describe('Profile-tag events', () => {
     cy.get('cx-profiletag');
     cy.wait(`@${homePage}`).its('status').should('eq', 200);
   });
+
   it('should send a CartChanged event on adding an item to cart', () => {
     goToProductPage();
     cy.get('cx-add-to-cart button.btn-primary').click();
     cy.get('cx-added-to-cart-dialog .btn-primary');
     cy.window().then((win) => {
-      expect((<any>win).Y_TRACKING.eventLayer.length).to.equal(2);
-      expect((<any>win).Y_TRACKING.eventLayer[1]['name']).to.equal(
-        'CartSnapshot'
-      );
-      const cartPayload = JSON.stringify(
-        (<any>win).Y_TRACKING.eventLayer[1]['data']
-      );
+      expect(
+        (<any>win).Y_TRACKING.eventLayer.filter(
+          (event) => event.name === cartSnapshotEventName
+        ).length
+      ).to.equal(1);
```

Comment:
Why does the length change here? I believe there should still be 2 events -> navigated + cartsnapshot

Reply:
I found that intermittently other events would make it into the layer, which could be carousel view events that got raised before profile tag loaded or something. So I thought I'd do what was done in most of the other tests where it filters the events based on the ones it cares about

Reply:
This is due to cypress sharing state sometime when it shouldn't. I really found no good way to get around this. Unfortunately sometimes we want to view the full array to make sure that it is the exact number of events we expect, but in this case this filtering is the right approach :+1: